src/mpid/ch4/netmod/ucx/ucx_pre.h

```diff
@@ -30,7 +30,11 @@ typedef struct {
 
 typedef struct {
     int handler_id;
-    char *pack_buffer;
+    ucp_datatype_t datatype;
+    union {
+        ucp_dt_iov_t *iov;
```

Comment:
In ch4/ofi we just statically allocate the iov array. We should probably do that here, too, and save the extra call to malloc in the communication path. E.g.

> ucp_dt_iov_t iov[2];

Reply:
This has to be retained in the heap. If I allocate it statically if the request does not complete immediately if grabs random data from the stack afterwards

Reply:
Nevermind I was looking at the wrong place in the code